studio/schemas/objects/compensations/bonusesByLocation.ts

```diff
@@ -49,47 +53,12 @@ export const bonusesByLocation = defineField({
     }),
   ],
   validation: (Rule) =>
-    Rule.custom((bonusesByLocation, context) => {
-      const duplicateCheck = checkForDuplicateLocations(
-        bonusesByLocation as BonusEntry[] | undefined,
+    Rule.custom((bonusesByLocation) => {
```

Comment:
I think we would introduce a lot of overhead and extra work if we stop trusting our own typing

Reply:
I agree. It's more a failsafe to avoid errors if someone adjust the checkForDuplicateLocations to return say `boolean | undefined`.

Given the fact that this repository can and will have every single developer at Variant add and edit code, it's a good thought to be "suspicious" of the typing in the repository.

Reply:
I don't think you agree then? 🙃 

If we assume that developers will come in and change function signatures without checking usages then we would need a lot of failsafes everywhere.

But we can add an extra check if that lets us merge this PR. My IDE complains about the redundancy in the suggestion above, so what about this?
```js
  validation: (Rule) =>
    Rule.custom((bonusesByLocation) => {
      const isNotDuplicate: boolean = checkForDuplicateLocations(
        bonusesByLocation as DocumentWithLocation[] | undefined,
      )
      return (
        isNotDuplicate ||
        "Each location should be listed only once in the bonuses list. You can assign the same bonus amount to multiple locations, but make sure no location appears more than once."
      );
    }),
```
Here we explicitly declare that the return type is expected to be `boolean`.

Reply:
No. Let me specify: I don't agree with your approach. I agree with your concern. 

We're not **assuming** that developers will come in and maintain our codebase, we **know** they will. That's the way it is today, and that's what the company intends for this project as well. 

I think your suggestion here is good.

Reply:
I didn't say we would not have new developers maintaining the codebase. But the trust we have in them is very low with this approach.

Will add the suggestion for now.

Reply:
It’s perfectly fine to have different opinions on this. However, this is not about doubting our developers’ intentions and expertise, but rather about being cautious of unintentional mistakes. Which can and often happens, regardless of level of experience. As a tech lead, I prefer to have such an approach. 😊